From c3b6f834fa73f1c7f38872ca46e7b9b84a736110 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Thu, 3 Sep 2020 12:37:37 -0700
Subject: [PATCH 1562/1852] v4l: xilinx: axis-switch: Remove an invalid if
 check condition

commit d15ffe75e2a9466f13e79e30a8ee615bb7ef1f7a from
https://github.com/Xilinx/linux-xlnx.git

drivers/media/platform/xilinx/xilinx-axis-switch.c:388:30: warning: comparison of unsigned expression < 0 is always false [-Wtype-limits]
  388 |  if (ret < 0 || routing_mode < 0 || routing_mode > 1) {
      |                              ^

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Reviewed-by: Vishal Sagar <vishal.sagar@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-axis-switch.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/xilinx/xilinx-axis-switch.c b/drivers/media/platform/xilinx/xilinx-axis-switch.c
index 86cd8585116d..c43578cc0c47 100644
--- a/drivers/media/platform/xilinx/xilinx-axis-switch.c
+++ b/drivers/media/platform/xilinx/xilinx-axis-switch.c
@@ -385,7 +385,7 @@ static int xvsw_parse_of(struct xvswitch_device *xvsw)
 	}
 
 	ret = of_property_read_u32(node, "xlnx,routing-mode", &routing_mode);
-	if (ret < 0 || routing_mode < 0 || routing_mode > 1) {
+	if (ret < 0 || routing_mode > 1) {
 		dev_err(xvsw->dev, "missing or invalid xlnx,routing property\n");
 		return ret;
 	}
-- 
2.31.1

